src/gt4py/stencil_builder.py

```diff
@@ -0,0 +1,210 @@
+import pathlib
+from typing import Callable, Optional, Dict, Any, Type, TypeVar, Union
+
+import gt4py
+from gt4py.definitions import BuildOptions, StencilID
+from gt4py.protocols import StencilFunc, AnnotatedStencilFunc
+from gt4py.ir import StencilDefinition, StencilImplementation
+
+
+SelfType = TypeVar("SelfType", bound="StencilBuilder")
+FrontendType = TypeVar("FrontendType", bound="gt4py.frontend.concepts.Frontend")
+BackendType = TypeVar("BackendType", bound="gt4py.backend.base.Backend")
+
+
+class StencilBuilder:
+    """
+    Orchestrates code generation and compilation.
+
+    Parameters
+    ----------
+    definition_func: Union[StencilFunc, AnnotatedStencilFunc]
```

Comment:
Again, I think that type annotations in comments should be removed if the code has been already annotated.